src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -13,6 +13,12 @@ public class StringFunctions {
     private static final char SPACE = ' ';
 
     public static final String reverseWithStringConcat(String string) {
+        if (string == null) {
+            return "";
+        }
+        if (string.isEmpty()) {
```

Comment:
Check for if a string that exists is empty, returning an empty string if it is empty.